assets/wizards/newspack/views/settings/connections/recaptcha.tsx

```diff
@@ -0,0 +1,171 @@
+			);
+		}
+	}, [ settings.use_captcha, settings.site_key, settings.site_secret ] );
+
+	useEffect( () => {
+		wizardApiFetch< RecaptchaData >(
+			{
+				path: '/newspack/v1/recaptcha',
+			},
+			{
+				onSuccess( fetchedSettings ) {
+					if ( fetchedSettings ) {
+						setSettings( fetchedSettings );
+						setSettingsToUpdate( { ...settingsDefault, ...fetchedSettings } );
+					}
+				},
+			}
+		);
+	}, [] );
+
+	function updateSettings( data: RecaptchaData ) {
+		wizardApiFetch< RecaptchaData >(
+			{
+				path: '/newspack/v1/recaptcha',
+				method: 'POST',
+				data,
+				updateCacheMethods: [ 'GET' ],
+			},
+			{
+				onSuccess( fetchedSettings ) {
+					setSettings( fetchedSettings );
+					setSettingsToUpdate( fetchedSettings );
+				},
+			}
+		);
+	}
+
+	return (
+		<>
+			<WizardsActionCard
+				isMedium
+				title={ __( 'Enable reCAPTCHA v3', 'newspack-plugin' ) }
+				description={ () => (
+					<Fragment>
+						{ isFetching && ! Boolean( settings.use_captcha ) ? (
+							__( 'Loading…', 'newspack-plugin' )
+						) : (
+							<>
+								{ __(
+									'Enabling reCAPTCHA v3 can help protect your site against bot attacks and credit card testing.',
+									'newspack-plugin'
+								) }{ ' ' }
+								<ExternalLink href="https://www.google.com/recaptcha/admin/create">
+									{ __( 'Get started', 'newspack-plugin' ) }
+								</ExternalLink>
+							</>
+						) }
+					</Fragment>
+				) }
+				hasGreyHeader={ !! settings.use_captcha }
+				toggleChecked={ !! settings.use_captcha }
+				toggleOnChange={ () => updateSettings( { use_captcha: ! settings.use_captcha } ) }
+				actionContent={
+					settings.use_captcha && (
+						<Button
+							variant="primary"
+							disabled={ isFetching || ! Object.keys( settingsToUpdate ).length }
+							onClick={ () => updateSettings( settingsToUpdate ) }
+						>
+							{ isFetching
+								? __( 'Loading…', 'newspack-plugin' )
+								: __( 'Save Settings', 'newspack-plugin' ) }
+						</Button>
+					)
+				}
+				error={ settings.use_captcha ? errorMessage : null }
+				disabled={ isFetching }
+			>
+				{ settings.use_captcha && (
+					<Fragment>
+						<Grid noMargin rowGap={ 16 }>
+							<TextControl
+								value={ settingsToUpdate?.site_key || '' }
+								label={ __( 'Site Key', 'newspack-plugin' ) }
+								onChange={ ( value: string ) =>
+									setSettingsToUpdate( { ...settingsToUpdate, site_key: value } )
+								}
+								disabled={ isFetching }
+								autoComplete="off"
+							/>
+							<TextControl
+								type="password"
+								value={ settingsToUpdate?.site_secret || '' }
+								label={ __( 'Site Secret', 'newspack-plugin' ) }
+								onChange={ ( value: string ) =>
+									setSettingsToUpdate( { ...settingsToUpdate, site_secret: value } )
+								}
+								disabled={ isFetching }
+								autoComplete="off"
+							/>
+							<TextControl
+								type="number"
+								step="0.05"
+								min="0"
```

Comment:
Question: Looks like it's possible to save this as 0 or empty, while on `trunk` this needs to be a non-zero value. I suspect we want to retain this non-zero behavior, but am not sure if there was some discussion around this. Is this an expected change?

Reply:
Good question... did a little digging.

Found that there's some code missing between `trunk` and this branch introduced in https://github.com/Automattic/newspack-plugin/commit/876d0a9a1baa26480e399d52199d62d8a8f6bed5#diff-1833fd29b967510299e86f14a13280961ec87bb2a1813386f741228cbc2b12fc. I have synced these changes.

**RE `threshold` value and the misalignment between presentation and storage on `trunk` i.e. _/wp-admin/admin.php?page=newspack-connections-wizard#/:_**

* The default value of `threshold` is [0.5](https://github.com/Automattic/newspack-plugin/blob/feat/ia-settings-connections__recaptcha/includes/class-recaptcha.php#L126)
* Attempting to save `0` will save `0` in the DB (Side note: the error with inputs clearing when saved has been fixed in this PR). `wp option get newspack_recaptcha_threshold` will yield `0`. Screenshot of Adminer below.
![Screenshot 2024-06-26 at 10 14 41](https://github.com/Automattic/newspack-plugin/assets/3676975/97a75c3b-67fd-4eb9-a5f5-0a19b32dd540)
* When a zero value is retrieved for the `newspack/v1/recaptcha` route it is evaluated as `empty()` and the default value of `0.5` is applied and sent.
![Screenshot 2024-06-26 at 10 22 03](https://github.com/Automattic/newspack-plugin/assets/3676975/b4454a80-a30d-4497-8956-267d13055697)

Enforcing a non-negative float shouldn't be too difficult. Let me know if you'd like me to proceed?

Reply:
Truthfully, I'm not too familiar with recaptcha, and so am not really sure what the expected behavior is here. Based on the `get_setting` code, it seems as though we always want a non-zero threshold (despite allowing this to be saved as zero). I would say let's enforce a non-negative, non-zero float, but lets let @dkoo chime in here since he is probably more familiar with the requirements.

Reply:
Hey guys, thanks for laying this all out. I think the intended behavior should be to enforce a non-zero float value, since `0` would effectively mean disabling reCAPTCHA entirely.

Reply:
Makes sense @dkoo , will update now. Will make `min="0.1"`

Reply:
Perfect, thanks!

Reply:
@chickenn00dle @dkoo I have added code introduced in #3126 and stringent validation for input fields in b7119362109bb0d9fe8857ee5932e526ad6336f4

A more verbose approach was necessary since only adding `min="0.1"` to the input field prevented values `<0.1` when using up/down arrow keys and up/down input UI ![Screenshot 2024-07-05 at 12 45 45](https://github.com/Automattic/newspack-plugin/assets/3676975/305b1091-18a7-4c63-8a51-68a108a093dc).

Reply:
> A more verbose approach was necessary since only adding min="0.1" to the input field prevented values <0.1 when using up/down arrow keys and up/down input UI Screenshot 2024-07-05 at 12 45 45.

Thanks for working on this @jaredrethman 

I'm a bit confused as to why we can't just set the min value here to `0.1`. It looks like the updated implementation returns an error anyway whenever we submit any value < 0.1 anyway, so we should probably just not let publishers input anything less than this in the first place. 

In any case, I'm happy to retain the additional validation you've added, but we should definitely still set this min value.